iree/turbine/kernel/wave/wave.py

```diff
@@ -204,6 +204,18 @@ def __init__(
     ):
         super().__init__(eager_function)
 
+        assert (
+            constraints is None
+            or next(
+                (
+                    c
+                    for c in constraints
+                    if isinstance(c, Constraint) and c.is_implicit()
```

Comment:
This is quite verbose, how about:
```
if isinstance(constraints, Sequence):
    assert all([not c.is_implicit for c in constraints if isinstance(c, Constraint)])
```
Are there cases where `constraints == None` or not specified? If not, we should omit that